src/common/utility_mac.mm

```diff
@@ -0,0 +1,263 @@
+        NSString *plistFile = [NSHomeDirectory() stringByAppendingFormat:@"/Library/LaunchAgents/%@.plist", appIdentifier];
+
+        // An error might occur in the code below, but we cannot report anything, so we just ignore them.
+
+        if ([fileManager fileExistsAtPath:plistFile]) {
+            auto maybePlist = readPlistFromFile(plistFile);
+            if (!maybePlist) {
+                // broken plist, overwrite it
+                auto result = writeNewPlistFile(plistFile, fullPath, enable);
+                if (!result) {
+                    qCWarning(lcUtility) << Q_FUNC_INFO << result.error();
+                }
+                return;
+            }
+            NSDictionary *plist = *maybePlist;
+
+            id programValue = plist[@"Program"];
+            if (programValue == nil) {
+                // broken plist, overwrite it
+                auto result = writeNewPlistFile(plistFile, fullPath, enable);
+                if (!result) {
+                    qCWarning(lcUtility) << result.error();
+                }
+            } else if (![fileManager fileExistsAtPath:programValue]) {
+                // Ok, a plist from some removed program, overwrite it
+                auto result = writeNewPlistFile(plistFile, fullPath, enable);
+                if (!result) {
+                    qCWarning(lcUtility) << result.error();
+                }
+            } else if ([fullPath compare:programValue options:NSCaseInsensitiveSearch] == NSOrderedSame) { // (Note: case insensitive compare, because most fs setups on mac are lse if (![fileManager fileExistscase insensitive)
+                // Wohoo, it's ours! Now carefully change only the RunAtLoad entry. If any value for
+                // e.g. KeepAlive was changed, we leave it as-is.
+                auto result = modifyPlist(plistFile, plist, enable);
+                if (!result) {
+                    qCWarning(lcUtility) << result.error();
+                }
+            } else if ([fullPath hasPrefix:@"/Applications/"]) {
+                // ok, we seem to be an officially installed application, overwrite the file
+                auto result = writeNewPlistFile(plistFile, fullPath, enable);
+                if (!result) {
+                    qCWarning(lcUtility) << result.error();
+                }
+            } else {
+                qCInfo(lcUtility) << "We're not an installed application, there is anoter executable "
+                                     "mentioned in the plist file, and that executable seems to exist, "
+                                     "so let's not touch the file.";
+            }
+        } else {
+            // plist doens't exist, write a new one.
+            auto result = writeNewPlistFile(plistFile, fullPath, enable);
+            if (!result) {
+                qCWarning(lcUtility) << result.error();
+            }
+        }
+    }
+}
+
+#ifndef TOKEN_AUTH_ONLY
```

Comment:
I think you should remove this
if `TOKEN_AUTH_ONLY` is defined, the build would most probably fail ?